packages/plugin-cosmos/src/services/fee-estimator.ts

```diff
@@ -0,0 +1,31 @@
+import { SigningCosmWasmClient } from "@cosmjs/cosmwasm-stargate";
+import { Coin, MsgSendEncodeObject, StdFee } from "@cosmjs/stargate";
+
+export class FeeEstimator {
+    private signingCosmWasmClient: SigningCosmWasmClient;
+    constructor(signingCosmWasmClient: SigningCosmWasmClient) {
+        this.signingCosmWasmClient = signingCosmWasmClient;
+    }
+
+    estimateGasForSendTokens(
+        senderAddress: string,
+        recipientAddress: string,
+        amount: readonly Coin[],
+        memo = ""
+    ): Promise<number> {
+        const sendMsg: MsgSendEncodeObject = {
+            typeUrl: "/cosmos.bank.v1beta1.MsgSend",
+            value: {
+                fromAddress: senderAddress,
+                toAddress: recipientAddress,
+                amount: [...amount],
+            },
+        };
+
+        return this.signingCosmWasmClient.simulate(
+            senderAddress,
+            [sendMsg],
+            memo
+        );
+    }
```

Comment:
the auto setting isnt enought for token transfers?

or it is required to calculate estimated gas fee before execute real transfer transaction to prevent redudant calls?

Reply:
I got some errors with `auto`